x-pack/plugins/security_solution/server/search_strategy/event_filters_fields/index.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { from } from 'rxjs';
+import type { ISearchStrategy, SearchStrategyDependencies } from '@kbn/data-plugin/server';
+import { IndexPatternsFetcher } from '@kbn/data-plugin/server';
+
+import {
+  formatIndexFields,
+  dedupeIndexName,
+  findExistingIndices,
+} from '@kbn/timelines-plugin/server/search_strategy/index_fields';
+
+import type {
+  BeatFields,
+  IndexField,
+  IndexFieldsStrategyRequest,
+  IndexFieldsStrategyResponse,
+} from '../../../common/search_strategy';
+import type { EndpointAppContextService } from '../../endpoint/endpoint_app_context_services';
+
+/**
+ * EventFiltersFieldProvider mimics indexField provider from timeline plugin: x-pack/plugins/timelines/server/search_strategy/index_fields/index.ts
```

Comment:
Just wondering: how much of this is the same as the function in timelines?  Would it make more sense to just convert theirs to a function that take in as input the ES Client? and then we can just re-use it from here by providing the internal ES client?

Reply:
Good call, will check this 👍

Reply:
Done [here](https://github.com/elastic/kibana/pull/145883/commits/13d998993f2e7eac2cac64b24c8698a87085e971), let me know what do you think